### ISSUE TITLE
Incorrect formatting due to caching

### ISSUE THREAD

**Issue by renkun-ken** [r-lib/styler#594]
```r
mtcars %>%
  f() # hello
```

is formatted to

```r
# hello
```

It won't happen if the comment is removed or `%>%` is replaced with `+`.

**Comment by lorenzwalthert** [comment]
This probably only happens under some circumstances. Please do not `cache_clear()` because then your problem won't be reproducible anymore I assume. Can you share the cache directory with me for debugging? You can find it with `cache_info()`.

**Comment by lorenzwalthert** [comment]
Here is a reprex:

``` r
library(styler)
cache_clear(ask = FALSE)
cache_activate()
#> Using cache 1.3.9000 at /Users/lorenz/Library/Caches/R/R.cache/styler/1.3.9000.
# style_text("f() # hello")
style_text("mtcars")
#> mtcars
style_text("mtcars %>%
  f()")
#> mtcars %>%
#>   f()

style_text("mtcars %>%
  f() # ")
#> #
```

<sup>Created on 2020-02-13 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by lorenzwalthert** [comment]
It seems as it is a duplicate of #592. At least the corresponding PR in #593 fixes it.

**Comment by lorenzwalthert** [comment]
Closed with #593.
